packages/asset-swapper/src/utils/market_operation_utils/index.ts

```diff
@@ -136,8 +148,14 @@ export class MarketOperationUtils {
             ...opts,
         };
         const [makerToken, takerToken] = getOrderTokens(nativeOrders[0]);
-        const [fillableAmounts, dexQuotes] = await this._sampler.executeAsync(
+        const [fillableAmounts, ethToTakerAssetRate, dexQuotes] = await this._sampler.executeAsync(
             DexOrderSampler.ops.getOrderFillableMakerAmounts(nativeOrders),
+            DexOrderSampler.ops.getMedianSellRate(
+                difference(FEE_QUOTE_SOURCES, _opts.excludedSources),
+                this._wethAddress,
+                makerToken,
+                ONE_ETHER,
+            ),
```

Comment:
This one is also wrong because it's actually getting maker -> ETH (we want ETH -> taker).